protocol/xmrmaker/claim.go

```diff
@@ -65,15 +68,21 @@ func (s *swapState) claimFunds() (ethcommon.Hash, error) {
 		// TODO: eventually update when relayer discovery is implemented
 		txHash, err = s.claimRelayer()
 		if err != nil {
-			return ethcommon.Hash{}, err
+			log.Warnf("failed to claim using relayer at %s, trying relayers via p2p: err: %s",
+				s.offerExtra.RelayerEndpoint,
+				err,
+			)
+			txHash, err = s.discoverRelayersAndClaim()
 		}
+	} else if ethBalance.Uint64() == 0 {
```

Comment:
The maximum amount of ETH that you can represent as wei in `uint64` is a little under 18.5. The documentation for `Uint64()` says the value is undefined if the value is too large.
```
$ python3 -c "print((2**64-1)/1E18)"
18.446744073709553
```